RecoMET/METAlgorithms/src/HcalHaloAlgo.cc

```diff
@@ -23,14 +23,11 @@ bool CompareTowers(const CaloTower* x, const CaloTower* y ){
   return x->iphi()*1000 + x->ieta() < y->iphi()*1000 + y->ieta();
 }
 
-HcalHaloAlgo::HcalHaloAlgo()
-{
+HcalHaloAlgo::HcalHaloAlgo() : geo_(nullptr), hgeo_(nullptr) {
```

Comment:
here and in the rest of this PR please implement the backport in a way identical to changes in #19910 (unless there was something after #19910 that added new changes currently in the master).
In this file I see that at least a half of the changes are not the same in text, even though they appear to be logically the same.